spec/features/inputs/select_filter_input_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+
+describe "Select Filter Input", type: :feature do
+  context 'when used with an enum column' do
+    let(:option_selector) { '.filter_form select[name="q[status_eq]"] option' }
+
+    context "with collection option" do
+      before do
+        register_page(Invoice) do
+          filter :status, as: :select, collection: [
+            ['Activo, si se considera', 0],
+            ['Archivado, ya no se considera', 1],
+            ['Estado que no existe', 42]
+          ]
+        end
+
+        visit admin_invoices_path
+      end
+
+      it "uses collection directly" do
+        expect(page.find("#{option_selector}[value='0']").text).to eq('Activo, si se considera')
+        expect(page.find("#{option_selector}[value='1']").text).to(
+          eq('Archivado, ya no se considera')
+        )
+        expect(page.find("#{option_selector}[value='42']").text).to eq('Estado que no existe')
+      end
```

Comment:
Maybe it would be good to test that the collection is actually filtered, what do you think? maybe it's not necessary

Reply:
I agree, especially considering that the filter only works with the db value, that way we make sure it's working. I'll add the test soon before merging